apps/explorer/src/components/Object/ObjectFieldsCard.tsx

```diff
@@ -1,65 +1,85 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-import { useGetObject } from '@mysten/core';
-import { useNormalizedMoveStruct } from '@mysten/dapp-kit';
 import { Search24 } from '@mysten/icons';
+import { type SuiMoveNormalizedStruct, type SuiObjectResponse } from '@mysten/sui.js/client';
 import { Text, LoadingIndicator, Combobox, ComboboxInput, ComboboxList } from '@mysten/ui';
-import { useState } from 'react';
+import clsx from 'clsx';
+import { useCallback, useEffect, useState } from 'react';
 
 import { FieldItem } from './FieldItem';
 import { ScrollToViewCard } from './ScrollToViewCard';
 import { getFieldTypeValue } from './utils';
+import { FieldsCard, FieldCollapsible, FieldsContainer } from '~/components/Object/FieldsUtils';
 import { Banner } from '~/ui/Banner';
-import { DisclosureBox } from '~/ui/DisclosureBox';
-import { TabHeader } from '~/ui/Tabs';
-import { ListItem, VerticalList } from '~/ui/VerticalList';
+import { Description } from '~/ui/Description';
+
+const DEFAULT_OPEN_FIELDS = 3;
+const DEFAULT_FIELDS_COUNT_TO_SHOW_SEARCH = 10;
 
 interface ObjectFieldsProps {
 	id: string;
+	normalizedStructData?: SuiMoveNormalizedStruct;
+	suiObjectResponseData?: SuiObjectResponse;
+	loading: boolean;
+	error: boolean;
+	objectType?: string;
 }
 
-export function ObjectFieldsCard({ id }: ObjectFieldsProps) {
-	const { data, isLoading, isError } = useGetObject(id);
+export function ObjectFieldsCard({
+	id,
+	normalizedStructData,
+	suiObjectResponseData,
+	loading,
+	error,
+	objectType,
+}: ObjectFieldsProps) {
 	const [query, setQuery] = useState('');
```

Comment:
Why dont you just call `const { data, isLoading, isError } = useGetObject(id)`  rather than passing it as a prop. same for `normalizedStructData`